op_crates/crypto/lib.rs

```diff
@@ -41,3 +48,43 @@ pub fn op_get_random_values(
 
   Ok(json!({}))
 }
+
+lazy_static! {
+  static ref DIGEST_ALGORITHMS_BY_NAME: HashMap<&'static str, &'static ring::digest::Algorithm> = {
+    let mut hm = HashMap::new();
+    hm.insert("SHA-1", &ring::digest::SHA1_FOR_LEGACY_USE_ONLY);
+    hm.insert("SHA-256", &ring::digest::SHA256);
+    hm.insert("SHA-384", &ring::digest::SHA384);
+    hm.insert("SHA-512", &ring::digest::SHA512);
+    hm
+  };
+}
+
+pub async fn op_subtle_digest(
+  _state: Rc<RefCell<OpState>>,
+  args: Value,
+  zero_copy: BufVec,
+) -> Result<Value, AnyError> {
+  let mut zc = zero_copy;
+
+  // FIXME: This chain of `unwrap` is meh.
```

Comment:
Is it deemed not clean enough, and if so how would you fix that?

Reply:
Users can still send garbage data manually, you use use something like `let args: SubtleDigestArgs = serde_json::from_value(args)?;` like it's done in other ops.

Reply:
> Users can still send garbage data manually

How? Can they call an OP directly without going through the JS API?

Regardless, I'll implement your suggestion.

Reply:
> Can they call an OP directly without going through the JS API?

Yes. This is not covered under any stability guarantee though, and is not officially documented. We should try minimize rust panics where possible and gracefully return errors to JS.

Reply:
> How? Can they call an OP directly without going through the JS API?

They can using JS API:
```js
Deno.core.dispatch("op_crypto", new TextEncoder().encode("some garbage data that is not valid json"));
```